src/v/cluster/archival_metadata_stm.cc

```diff
@@ -727,6 +718,8 @@ archival_metadata_stm::sync(model::timeout_clock::duration timeout) {
                   "Replication failed for archival STM command: {}",
                   res.error());
                 co_return false;
+            } else {
+                _last_replicate = std::nullopt;
```

Comment:
This probably needs to be pulled out of the else branch and have an additional test.